java/core/src/main/java/com/whylogs/core/metrics/Metric.java

```diff
@@ -19,6 +20,20 @@ public abstract class Metric {
 
   public abstract OperationResult columnarUpdate(PreprocessedColumn data);
 
+  public abstract HashMap<String, MetricComponent> getComponents();
+
+  public Metric merge(Metric other) {
+    Metric merged = this;
+    if (!this.namespace.equals(other.namespace)) {
+      throw new IllegalArgumentException("Cannot merge metrics with different namespaces");
+    }
+
+    if (this instanceof IntegralMetric) {
```

Comment:
This frightens me even more :(

Reply:
Let's chat about this. I don't like it either, but we need some way of knowing if this is an IntegralMetric or not. We could use the namespace. Sadly there are a couple places we use this where we loose the type info even with the CRTP.

Reply:
@naddeoa could you take a look at this section of the code? I can walk you through it, but basically we are using CRTP to be able to have a collection of Metrics but not loose the typing. Sadly there is a section we use merge on a Collection of Metrics ... which means we have type erasure. All this method is doing is figuring out the type and sending to the correct subclass merge. I'm all ears for a better way to do this.